dpnp/tests/test_utils.py

```diff
@@ -0,0 +1,67 @@
+import dpctl.tensor as dpt
+import numpy
+import pytest
+
+import dpnp
+
+
+class TestIsSupportedArrayOrScalar:
+    @pytest.mark.parametrize(
+        "array",
+        [
+            dpnp.array([1, 2, 3]),
+            dpnp.array(1),
+            dpt.asarray([1, 2, 3]),
+        ],
+    )
+    def test_valid_arrays(self, array):
+        assert dpnp.is_supported_array_or_scalar(array)
+
+    @pytest.mark.parametrize(
+        "value",
+        [
+            42,
+            True,
+            "1",
+        ],
+    )
+    def test_valid_scalars(self, value):
+        assert dpnp.is_supported_array_or_scalar(value)
+
+    @pytest.mark.parametrize(
+        "array",
+        [
+            [1, 2, 3],
+            (1, 2, 3),
+            None,
+            numpy.array([1, 2, 3]),
+        ],
+    )
+    def test_invalid_arrays(self, array):
+        assert not dpnp.is_supported_array_or_scalar(array)
+
+
+class TestSynchronizeArrayData:
+    @pytest.mark.parametrize(
+        "array",
+        [
+            dpnp.array([1, 2, 3]),
+            dpt.asarray([1, 2, 3]),
+        ],
+    )
+    def test_synchronize_array_data(self, array):
+        try:
+            dpnp.synchronize_array_data(array)
```

Comment:
It would be better to submit something to the queue before calling synchronization.